### ISSUE TITLE
How to pass variables with spaces?

### ISSUE THREAD

**Issue by VMM-Mtech** [ppodgorsek/docker-robot-framework#194]
Hi!

Running docker on windows. Following is my simple test case:

```
*** Test Cases ***
Test
    Should Be Equal  This with spaces  ${MY_VARIABLE}
```

If I run the test directly with robot like this, it succeeds:

`robot --variable MY_VARIABLE:"This with spaces" robot`

But if I try to run it with your docker image, it fails

`docker run -v %cd%/reports:/opt/robotframework/reports:Z -v %cd%/robot:/opt/robotframework/tests:Z -e ROBOT_OPTIONS="--loglevel DEBUG --extension robot -v MY_VARIABLE:\"This with spaces\" " my_docker_image`

The error message is "[ ERROR ] Parsing 'with' failed: File or directory to execute does not exist."

I've tried with singe quotes, single quotes escaped etc, and I just can't get this to work.

**Comment by VMM-Mtech** [comment]
`-E space:_` and using _ instead of spaces seems to work, but this is deprecated. That's why I created this issue

**Comment by UltimateDogg** [comment]
Can you try MY_VARIABLE:This\ with\ spaces.   ?

**Comment by VMM-Mtech** [comment]
`docker run -v %cd%/reports:/opt/robotframework/reports:Z -v %cd%/robot:/opt/robotframework/tests:Z -e ROBOT_OPTIONS="--loglevel DEBUG --extension robot -v MY_VARIABLE:This\ with\ spaces" my_docker_image
[ ERROR ] Parsing 'with\' failed: File or directory to execute does not exist.`

**Comment by UltimateDogg** [comment]
hmm, how about 
`docker run -v %cd%/reports:/opt/robotframework/reports:Z -v %cd%/robot:/opt/robotframework/tests:Z -e ROBOT_OPTIONS='--loglevel DEBUG --extension robot -v MY_VARIABLE:"This with spaces"'`

**Comment by VMM-Mtech** [comment]
Even worse.

`docker: invalid reference format: repository name must be lowercase.`

And as far as I know, we can't just drop double quotes from ROBOT_OPTIONS since we want to inject some variables there

**Comment by UltimateDogg** [comment]
https://stackoverflow.com/questions/56059255/docker-environment-variable-with-spaces

Ok this leads me to believe if even json notation doesnt work we dont have the correct escaping in the docker container itself

**Comment by VMM-Mtech** [comment]
If I looked at your code correctly, docker will run the run-tests-in-virtual-screen.sh script, right? Could you test if you could do some escaping or something there?

**Comment by Noordsestern** [comment]
I think, i did this once:
`robot -v "USERNAME:John Doe" greet_user.robot`
Exclamation marks include the variable.

**Comment by VMM-Mtech** [comment]
`docker run -v %cd%/reports:/opt/robotframework/reports:Z -v %cd%/robot:/opt/robotframework/tests:Z -e ROBOT_OPTIONS="--loglevel DEBUG --extension robot -v \"MY_VARIABLE:This with spaces\"" my_docker_image`

`[ ERROR ] Parsing 'with' failed: File or directory to execute does not exist.`

**Comment by UltimateDogg** [comment]
after several hours and about 50 combinations, i could not find a way to get it to work, even changing the .sh file. I think we could it to work by splitting each option into a separate -e but that is a lot of work and makes this very brittle. space:_ actually did work however but thats deprecated as you said. Maybe its time to ask for that to stay.

**Comment by VMM-Mtech** [comment]
Hi,

refreshing this one. The escape (or -E) worked with 3.6.0, but does not work at all with 3.7.0. 

```
docker run -v %cd%/reports:/opt/robotframework/reports:Z -v %cd%/robot:/opt/robotframework/tests:Z -e ROBOT_OPTIONS="--loglevel DEBUG --extension robot -v MY_VARIABLE:This_with_spaces -E space:_" ppodgorsek/robot-framework:3.7.0
[ ERROR ] option -E not recognized
```

I guess there is no new options for me, but to use older versions? I'd like to upgrade to the newest since there are some random issues occuring between geckodriver and firefox that hopefully would work better in newer versions...

**Comment by PapsOu** [comment]
For informations, I had the same problème by setting a custom user-agent.

TL;DR; : You should double quote the `$@` in `bin/chromium-browser.sh` and `bin/chromedriver.sh` files : 

```bash
#!/bin/sh

exec /usr/lib/chromium/chrome-original --disable-gpu --no-sandbox "$@"
```

Here my question (and the working solution too) : https://stackoverflow.com/questions/65995818/robot-framework-calling-add-argument-with-spaces

**Comment by UltimateDogg** [comment]
Good catch, let me submit and see if that works

**Comment by ppodgorsek** [comment]
The PR has been merged but reopening this issue, as we need to add automated tests.

**Comment by VMM-Mtech** [comment]
I'm not convinced that the fix PapsOu suggested will solve the whole issue since my issue was with RF itself and not the chrome in the container.

**Comment by VMM-Mtech** [comment]
Updating the situation:

We finally were forced to update to v. 3.8.0, where we couldn't use the -E flag anymore. So our workaround now is to pass the variables with underscores to our Robot scripts, and then replace the underscores with spaces for each variable before they are used. Not the most elegant solution, but luckily we only have a couple of variables and they are used just once, so the replacement needs to happen only in couple of places.

But as @UltimateDogg said in https://github.com/ppodgorsek/docker-robot-framework/issues/194#issuecomment-528122598, I guess the original problem still remains.
